### PR TITLE
Update cropping.js

### LINKED ISSUE #29: [Bug] TypeError: "gravity" is read-only
### **Describe the bug**

When trying to use a custom `crop` without setting a gravity, you get an error `TypeError: "gravity" is ready-only`

### **Is this a regression?**

Unsure

### **Steps To Reproduce the error**

```
<CldImage width="..." height="..." src="..." crop="crop" />
```

### **Expected behaviour**

Gravity should default to `auto` when not set

### **Media proof (Screenshot)**

image

### **Additional context**

Tests should be added for the cropping plugin

### PR DESCRIPTION
**Pull request by Smartmind12**
## Description
Fixes the Gravity issue while cropping image.

---
## Issue Ticket Number
Fixes #29 

---
## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

---
# Checklist:
- [X] I have followed the contributing guidelines of this project as mentioned in [CONTRIBUTING.md](/CONTRIBUTING.md)
- [X] I have created an [issue](https://github.com/colbyfayock/next-cloudinary/issues) ticket for this PR
- [X] I have checked to ensure there aren't other open [Pull Requests](https://github.com/colbyfayock/next-cloudinary/pulls) for the same update/change?
- [X] I have performed a self-review of my own code
- [X] I have run tests locally to ensure they all pass
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes needed to the documentation

### CODE REVIEW

**Comment by colbyfayock** [comment]
hey @Smartmind12 looks like we're still getting 2 failing tests, both URLs appear to be missing the `h_600` in the tests

File: next-cloudinary/src/plugins/cropping.js [hunk 1]
```diff
@@ -8,21 +8,21 @@ export function plugin({ cldImage, options, cldOptions } = {}) {
 
   let transformationString = `c_${crop},w_${width}`;
 
-  if ( !gravity && cropsGravityAuto.includes(crop) ) {
+  if ( !gravity ) {
+    if ( cropsGravityAuto.includes(crop) && !gravity ) {
```
Comment (colbyfayock) [inline next-cloudinary/src/plugins/cropping.js:12]: you can probably get rid of the additional `!gravity` check here as at this point, we're already in an `if` clause where the case is that we dont have gravity set

File: next-cloudinary/src/plugins/cropping.js [hunk 2]
```diff
@@ -8,21 +8,21 @@ export function plugin({ cldImage, options, cldOptions } = {}) {
...
     gravity = 'auto';
+    } else {
```
Comment (colbyfayock) [inline next-cloudinary/src/plugins/cropping.js:14]: you added this as an else related to the gravity configurations, but this isn't related to gravity. this is likely the cause of the regression

```
if ( !['limit'].includes(crop) ) {
    transformationString = `${transformationString},h_${height}`;
  }
```

that likely shouldn't be change at all

**Comment by colbyfayock** [comment]
@Smartmind12 can you please also write a test case to capture the gravity bug to make sure that it's satisfied and doesnt break?

File: next-cloudinary/src/plugins/cropping.js (hunk 2 above)
Comment (Smartmind12) [reply next-cloudinary/src/plugins/cropping.js:18]: I have added the Command again to the commit!

File: next-cloudinary/src/plugins/cropping.js (hunk 1 above)
Comment (colbyfayock) [reply next-cloudinary/src/plugins/cropping.js:16]: i was just suggesting you could instead do this:
```
  if ( !gravity ) {
    if ( cropsGravityAuto.includes(crop)) {
```

as the additional `!gravity` on the second line is redundant :) but not a big deal

Comment (Smartmind12) [reply next-cloudinary/src/plugins/cropping.js:16]: That makes sense!! I will change it!

**Comment by colbyfayock** [comment]
@Smartmind12 a test is still failing, please run `yarn test` locally inside of the `next-cloudinary`  directory

**Comment by Smartmind12** [comment]
> @Smartmind12 can you please also write a test case to capture the gravity bug to make sure that it's satisfied and doesnt break?

Kindly help me out with it because i would need guidance on how to do it!

**Comment by colbyfayock** [comment]
additionally I'm still seeing the error 
image

**Comment by colbyfayock** [comment]
@Smartmind12 let me know how you're feeling about this ticket - someone created a separate pull request for this: https://github.com/colbyfayock/next-cloudinary/pull/32

while you get first attempt at the PR, we can move forward with the other one if you're having trouble

thanks

**Comment by colbyfayock** [comment]
> > @Smartmind12 can you please also write a test case to capture the gravity bug to make sure that it's satisfied and doesnt break?
> 
> Kindly help me out with it because i would need guidance on how to do it!

here's an example of a test for the Raw Transformation plugin:

https://github.com/colbyfayock/next-cloudinary/blob/main/next-cloudinary/tests/plugins/raw-transformations.spec.js

you would need to create a new cropping.spec.js file and include a test like that but testing this functionality

**Comment by Smartmind12** [comment]
> @Smartmind12 let me know how you're feeling about this ticket - someone created a separate pull request for this: #32
> 
> while you get first attempt at the PR, we can move forward with the other one if you're having trouble
> 
> thanks

I checked out the other PR and i noticed no change as been made as compared to the original file.

**Comment by Smartmind12** [comment]
> additionally I'm still seeing the error image

This is error not pertaining to Gravity of the Image ...Check Line 46

**Comment by colbyfayock** [comment]
> > @Smartmind12 let me know how you're feeling about this ticket - someone created a separate pull request for this: #32
> > while you get first attempt at the PR, we can move forward with the other one if you're having trouble
> > thanks
> 
> I checked out the other PR and i noticed no change as been made as compared to the original file.

there are changes - instead of using `gravity` destructured from the `cldOptions` object, it's now referencing `cldOptions` directly

this is because `gravity` is defined as a `const` meaning it's read only (hence the error) where the property of `cldOptions` is mutable

**Comment by colbyfayock** [comment]
> > additionally I'm still seeing the error image
> 
> @colbyfayock This is error not pertaining to Gravity of the Image ...Check Line 46 It is different as compared to image shown in the Issue opened : image

it's the same error, the code around it has just changed. the arrow is pointing to the same code and its the same error message

**Comment by colbyfayock** [comment]
have you tested this locally? I checked out the branch and it doesnt work

**Comment by Smartmind12** [comment]
> > > additionally I'm still seeing the error image
> > 
> > 
> > @colbyfayock This is error not pertaining to Gravity of the Image ...Check Line 46 It is different as compared to image shown in the Issue opened : image
> 
> it's the same error, the code around it has just changed. the arrow is pointing to the same code and its the same error message

Yeaa my bad i noticed it and now i have tried updating it.

**Comment by colbyfayock** [comment]
it looks like that resolved the gravity error but i'm still seeing a regression in the tests

image


can you please:
- Fix the regression cause by your code change
- Add a new test to cover your code change

**Comment by Smartmind12** [comment]
> it looks like that resolved the gravity error but i'm still seeing a regression in the tests
> 
> image
> 
> can you please:
> 
> * Fix the regression cause by your code change
> * Add a new test to cover your code change

Yeaa i will lookinto it and add the test too!

**Comment by colbyfayock** [comment]
hi @Smartmind12 any luck? if this isn't wrapped up by early next week i'm going to pass this on to the other person wanting to contribute to this

**Comment by Smartmind12** [comment]
> hi @Smartmind12 any luck? if this isn't wrapped up by early next week i'm going to pass this on to the other person wanting to contribute to this

Heyy i was busy this week so will wrap it up by this saturday or early sunday!

**Comment by colbyfayock** [comment]
great thanks!

**Comment by colbyfayock** [comment]
image

**Comment by Smartmind12** [comment]
@colbyfayock do you want me to create a new test page or add it in the existing test page?

**Comment by colbyfayock** [comment]
yes please create a new file here: https://github.com/colbyfayock/next-cloudinary/tree/main/next-cloudinary/tests/plugins

**Comment by colbyfayock** [comment]
@Smartmind12 this still isn't working and is now throwing a new error. are you testing these  changes?

image

as i mentioned before, given the timeline, i'm going to move on to seeing if the other person wants to continue the work on their branch

thank you for your attempt to get this working

**Comment by Smartmind12** [comment]
> as i mentioned before, given the timeline, i'm going to move on to seeing if the other person wants to continue the work on their branch
> 
> thank you for your attempt to get this working

@colbyfayock I am sorry for the delay but I have updated the changes and created a test too for the same. Kindly review it and let me know if it works!

**Pull request closed without merging**